catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/app/UserApplication.java

```diff
@@ -284,15 +289,16 @@ private Map<String, Object> mapAttributes(Map<String, Object> persistentItem) {
 
   private Map<String, Object> getSubjectAttributes(Subject subject) {
     // @formatter:off
-    Map<String, Object> required =
-        ImmutableMap.of(
-            "userid", subjectIdentity.getUniqueIdentifier(subject),
-            "username", SubjectUtils.getName(subject),
-            "isGuest", subject.isGuest(),
-            "roles", getSubjectRoles(subject),
-            "preferences", getSubjectPreferences(subject));
+    Map<String, Object> required = new HashMap<>();
+    required.put("userid", subjectIdentity.getUniqueIdentifier(subject));
+    required.put("username", SubjectUtils.getName(subject));
+    required.put("isGuest", subject.isGuest());
+    required.put("roles", getSubjectRoles(subject));
+    required.put("preferences", getSubjectPreferences(subject));
     // @formatter:on
 
+    attributesModifiers.forEach(modifier -> modifier.modifyAttributes(required, subject));
+
     String email = SubjectUtils.getEmailAddress(subject);
 
     if (StringUtils.isEmpty(email)) {
```

Comment:
Do we want this to return something immutable?  If so, do we want to change the line inside here to be return ImmutableMap.<String, Object>builder().putAll(required) ?

Reply:
I don't think it matters, since the map returned by this method is provided directly as the endpoint response, but I went ahead and used `ImmutableMap.copyOf()`.